### DISCUSSION TITLE
How to generate table in storybook .mdx  files

### DISCUSSION THREAD

**Discussion by JohnVersus** [storybookjs/storybook#22613] [Help] [answered]
I have table added in the .md format like below 
```
| Prop         | CSS Property |
| ------------ | ------------ |
| `boxShadow`  | box-shadow   |
| `textShadow` | text-shadow  |
```

But storybook is not rendering the table. This is how it is rendered. 
```
| Prop | CSS Property | | ------------ | ------------ | | boxShadow | box-shadow | | textShadow | text-shadow |

```  

Any suggestions for adding a table?

Thank you

**Comment by f3bw** [comment, marked as answer]
You could try a more complex one (extra styling etc), something like this:

jsx
```
import React from 'react';

const Table = ({ headers, rows }) => (
  <table>
    <thead>
      <tr>
        {headers.map((header, index) => (
          <th key={index}>{header}</th>
        ))}
      </tr>
    </thead>
    <tbody>
      {rows.map((row, rowIndex) => (
        <tr key={rowIndex}>
          {row.map((cell, cellIndex) => (
            <td key={cellIndex}>{cell}</td>
          ))}
        </tr>
      ))}
    </tbody>
  </table>
);

export default Table;
```

Markdown:
```
import Table from './Table';

# My Storybook Documentation

Here's an example table:

<Table
  headers={['Header 1', 'Header 2']}
  rows={[
    ['Cell 1', 'Cell 2'],
    ['Cell 3', 'Cell 4'],
  ]}
/>

```

And then of course change it to your liking.

**Reply by JohnVersus** [reply]
Thanks, @fabricewerger 

This trick made my code much cleaner.

**Reply by MrXwq** [reply]
Do I have to write it myself? Why no longer support the old way of writing md

**Comment by BHerbulot** [comment]
You can import **Markdown** from '@storybook/blocks' and put your table inside

```
<Markdown>
{`
| Prop         | CSS Property |
| ------------ | ------------ |
| `boxShadow` | box-shadow   |
| `textShadow` | text-shadow  |
`}
</Markdown>
```

**Reply by terrymun** [reply]
@BHerbulot Thanks for the reply! This is the solution that currently works for us: for some reason we simply cannot get remarkGfm to work with our storybook: it keeps complaining about an ESM import issue.

**Reply by olivierr91** [reply]
Cannot get this to work, I get a `Could not parse expression with acorn` error.

**Comment by AnsonH** [comment]
Storybook officially recommends enabling the [remark-gfm](https://github.com/remarkjs/remark-gfm) plugin in the configuration file:

1. Install `remark-gfm` as dev dependency
2. Follow [this guide](https://storybook.js.org/docs/writing-docs/mdx#markdown-tables-arent-rendering-correctly)
